substrate/frame/democracy/src/lib.rs

```diff
@@ -1190,6 +1197,74 @@ pub trait EncodeInto: Encode {
 }
 impl<T: Encode> EncodeInto for T {}
 
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// This should be valid before or after each state transition of this pallet.
+	pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+		// PublicPropCount should be greater than PublicProps count
+		ensure!(
+			PublicProps::<T>::get().len() <= <PublicPropCount<T>>::get() as usize,
+			"`PublicPropCount` should be greater than of `PublicProps` in storage"
+		);
+		// DepositOf should be equal to PublicProps in storage
+		ensure!(
+			DepositOf::<T>::iter().count() == PublicProps::<T>::get().len() as usize,
+			"`DepositOf` should be greater than of `PublicProps` in storage"
+		);
+		// Total number of depositors should be greater or equal to `PublicProps`
+		ensure!(
+			Self::count_total_depositors() >= PublicProps::<T>::get().len() as u32,
+			"Total depositors should be greater than that of `PublicProps` in storage"
+		);
+		// Cancellations should be less than or equal to ReferendumInfoOf
+		ensure!(
+			Cancellations::<T>::iter().count() <= ReferendumInfoOf::<T>::iter().count() as usize,
+			"`Cancellations` should be less than or must equal `ReferendumInfoOf` in storage"
+		);
```

Comment:
When cancelling a referendum with the call `cancel_referendum` we remove the data from `ReferendumInfoOf`, I am not sure about this property.

Reply:
should i remove this invariant check?